packages/api-client-core/src/operationBuilders.ts

```diff
@@ -175,31 +167,32 @@ export const actionResultOperation = <Action extends AnyActionFunction, Options
   options?: Options
 ) => {
   let fields: FieldSelection = {};
+  let operationName = action.operationName;
 
   switch (action.type) {
     case "action": {
       const selection = options?.select || action.defaultSelection;
+      // background bulk actions enqueue many of the same action, each returning the result of one element of the bulk. so, the GraphQL result type is the singular result type, not the bulk result type.
+      if (action.isBulk) {
+        operationName = action.operationName.replace(/^bulk/, "").replace(/s$/, "");
+      }
+      const resultType = `${camelize(operationName)}Result`;
 
       fields = {
-        [`... on ${camelize(action.operationName)}Result`]: actionResultFieldSelection(
-          action.modelApiIdentifier,
-          selection,
-          action.isBulk,
-          action.hasReturnType
-        ),
+        [`... on ${resultType}`]: actionResultFieldSelection(action.modelApiIdentifier, selection, action.hasReturnType),
       };
       break;
     }
     case "globalAction": {
       fields = {
-        [`... on ${camelize(action.operationName)}Result`]: globalActionFieldSelection(),
+        [`... on ${camelize(operationName)}Result`]: globalActionFieldSelection(),
       };
     }
   }
 
   const actionResultOperation: BuilderOperation = {
     type: "subscription",
-    name: action.operationName,
+    name: capitalizeIdentifier(operationName) + "BackgroundResult",
```

Comment:
nit: This is in a function called `actionResultOperation`, but this change implies it's a background action. Given other stuff in here, does seems like this change is the right one, so perhaps we rename the function to be the more explicit `backgroundActionResultOperation`?